### ISSUE TITLE
Android 8.1: 411 connection failed, Owncloud

### ISSUE THREAD

**Issue by julianpoemp** [buttercup/buttercup-mobile#76]
When I try to connect buttercup with my owncloud I get the message: "Error: Connection failed, Invalid response: 411 undefined".

But the connection works using the desktop clients.

**Comment by perry-mitchell** [comment]
Hi @julianpoemp - Are you using just the ownCloud URL and not using the `remote.php/webdav` suffix? Does it work when you use the webdav connection on android?

**Comment by julianpoemp** [comment]
Hi @perry-mitchell, thanks for your fast answer.
The error message appear both using webDav or owncloud.
It still does not wok adding 'remote.php/webdav'.

Android: 8.0.0 (bought a new phone)
Name: Huaei Mate Pro 10
Buttercup: 1.1.2
Owncloud: 10.0.4

I tried:

- WebDav with URL https://upload.<myowncloud>.net/remote.php/webdav
- Owncloud with URL: https://upload.<myowncloud>.net

Connection via owncloud on the desktop client works all fine.

**Comment by perry-mitchell** [comment]
Is there any chance you could create a temporary account for me to test with? I’m available on keybase as perrymitchell so you could share the credentials there. Doing so would help me debug faster.

**Comment by julianpoemp** [comment]
yes, I can send you a test account. I send you a message in the next 10 minutes.

**Comment by perry-mitchell** [comment]
Ok, after the first round of testing:
 * On Android, Bcup (on this server) returns 411 Length Required when **connecting**
 * The connection works on iOS

**Comment by perry-mitchell** [comment]
@julianpoemp Next port of call is your web server.. Something here seems out of place - are you running any weird setup there? The 411 seems to come from a server where chunked encoding is not supported. What sort of setup are you running?

 * Web server (apache/nginx)
 * Php setup (fpm/mod-php)
 * Linux/Server operating system/distro

**Comment by julianpoemp** [comment]
@perry-mitchell 
I have owncloud installed on a webspace from www.all-inkl.com. It's a Linux system that uses an Apache server. PHP version is 7.0. Let me know if you need more information.

**Comment by julianpoemp** [comment]
Probably it's important to say, that I'm using a Let's Encrypt certificate.

**Comment by perry-mitchell** [comment]
Thanks. At this stage it looks to be a compatibility issue between react native and the service you’re using. When connecting, we use RN’s built in fetch function to make the request. If this error is occurring there, it means that the implementation on android is using a chunked transfer encoding - and the server doesn’t support it. Mine does, as well as all of the Nextcloud servers I’ve tested during the making of the app.

I would hope that this could be addressed by updating react native, which is soon due, but i don’t have any ideas for a quick fix sorry..

**Comment by julianpoemp** [comment]
Ok, thank you. After my exams I will update my owncloud installation und look for a workaround. If I found samething I let you know.

**Comment by julianpoemp** [comment]
@perry-mitchell , this bug is gone on Android with Buttercup Version 1.2.0. Thank you! 😄

**Comment by perry-mitchell** [comment]
Hey that’s excellent news!! Thanks for trying it out.. 😁🎉
